apps/test/unit/StudioAppTest.js

```diff
@@ -16,42 +16,41 @@ import project from '@cdo/apps/code-studio/initApp/project';
 import {sandboxDocumentBody} from '../util/testUtils';
 import sampleLibrary from './code-studio/components/libraries/sampleLibrary.json';
 import {createLibraryClosure} from '@cdo/apps/code-studio/components/libraries/libraryParser';
+import * as utils from '@cdo/apps/utils';
 
 describe('StudioApp', () => {
   sandboxDocumentBody();
 
   describe('StudioApp.singleton', () => {
-    beforeEach(stubStudioApp);
-    afterEach(restoreStudioApp);
-
     let containerDiv, codeWorkspaceDiv;
+
     beforeEach(() => {
+      stubStudioApp();
+      stubRedux();
+      registerReducers(commonReducers);
+
       codeWorkspaceDiv = document.createElement('div');
       codeWorkspaceDiv.id = 'codeWorkspace';
       document.body.appendChild(codeWorkspaceDiv);
-
       containerDiv = document.createElement('div');
       containerDiv.id = 'foo';
       containerDiv.innerHTML = `
-  <button id="runButton" />
-  <button id="resetButton" />
-  <div id="visualizationColumn" />
-  <div id="toolbox-header" />
-  `;
+      <button id="runButton" />
+      <button id="resetButton" />
+      <div id="visualizationColumn" />
+      <div id="toolbox-header" />
+      `;
       document.body.appendChild(containerDiv);
     });
 
     afterEach(() => {
+      restoreStudioApp();
+      restoreRedux();
+
       document.body.removeChild(codeWorkspaceDiv);
       document.body.removeChild(containerDiv);
     });
 
-    beforeEach(() => {
-      stubRedux();
-      registerReducers(commonReducers);
-    });
-    afterEach(restoreRedux);
-
```

Comment:
all changes above this line are just clean up. there were 3 separate beforeEach and afterEach statements for the same scope of tests, so i condensed them.